src/statsmodels/statsmodel.jl

```diff
@@ -48,8 +48,8 @@ for (modeltype, dfmodeltype) in ((:StatisticalModel, DataFrameStatisticalModel),
                                  (:RegressionModel, DataFrameRegressionModel))
     @eval begin
         function StatsBase.fit{T<:$modeltype}(::Type{T}, f::Formula, df::AbstractDataFrame,
-                                              args...; kwargs...)
-            mf = ModelFrame(f, df)
+                                              args...; contrasts::Dict = Dict(), kwargs...)
```

Comment:
I see the point there, and agree there's really no benefit to allowing `HelmertCoding` etc. I've changed the contrasts matrix method for `Type{C<:AbstractContrasts}` to give an informative error message.

Let's restrict this dict to `AbstractContrasts` for now and then see if other use cases pop up. I can imagine someone might want to construct a `ContrastsMatrix` directly (perhaps re-using from a previously fitted model's `ModelFrame`) but my intuition is to keep things simple here and introduce more complexity as needed.